contracts/kleros/KlerosLiquid.sol

```diff
@@ -0,0 +1,724 @@
+    }
+    struct Dispute {
+        uint subcourtID; // The ID of the subcourt the dispute is in.
+        Arbitrable arbitrated; // The arbitrated arbitrable contract.
+        uint choices; // The number of choices jurors have when voting.
+        Period period; // The current period of the dispute.
+        uint lastPeriodChange; // The last time the period was changed.
+        Vote[][] votes; // The votes in the form `votes[appeal][voteID]`.
+        VoteCounter[] voteCounters; // The vote counters in the form `voteCounters[appeal]`.
+        uint[] jurorAtStake; // The amount of PNK at stake for each juror in the form `jurorAtStake[appeal]`.
+        uint[] totalJurorFees; // The total juror fees paid in the form `totalJurorFees[appeal]`.
+        uint[] appealDraws; // The next voteIDs to draw in the form `appealDraws[appeal]`.
+        uint[] appealCommits; // The number of commits in the form `appealCommits[appeal]`.
+        uint[] appealVotes; // The number of votes in the form `appealVotes[appeal]`.
+        uint[] appealRepartitions; // The next voteIDs to repartition tokens/eth for in the form `appealRepartitions[appeal]`.
+    }
+
+    // Juror
+    struct Juror {
+        uint[] subcourtIDS; // The IDs of subcourts where the juror has activation path ends.
+        mapping(uint => bool) currentSubcourtIDSMap; // Map for efficient lookups of the subcourt IDs list.
+        mapping(uint => bool) subcourtIDSMap; // Map for efficient lookups of the subcourt IDs list.
+        uint atStake; // The juror's total amount of PNK at stake in disputes.
+    }
+
+    /* Events */
+
+    /** @dev Emitted when we pass to a new phase.
+     *  @param phase The new phase.
+     */
+    event NewPhase(Phase phase);
+
+    /** @dev Emitted when a dispute passes to a new period.
+     *  @param period The new period.
+     */
+    event NewPeriod(uint indexed disputeID, Period period);
+
+    /** @dev Emitted when a juror is drawn.
+     *  @param disputeID The ID of the dispute.
+     *  @param arbitrable The arbitrable contract that is ruled by the dispute.
+     *  @param _address The drawn address.
+     *  @param _voteID The vote ID.
+     */
+    event Draw(uint indexed disputeID, Arbitrable indexed arbitrable, address indexed _address, uint _voteID);
+
+    /** @dev Emitted when a juror wins or loses tokens and ETH from a dispute.
+     *  @param disputeID The ID of the dispute.
+     *  @param _address The juror affected.
+     *  @param tokenAmount The amount of tokens won or lost.
+     *  @param ETHAmount The amount of ETH won or lost.
+     */
+    event TokenAndETHShift(uint indexed disputeID, address indexed _address, int tokenAmount, int ETHAmount);
+
+    /* Storage */
+
+    // General Constants
+    uint public constant NON_PAYABLE_AMOUNT = (2 ** 256 - 2) / 2;
+    uint public constant ALPHA_DIVISOR = 1e4;
+    // General Contracts
+    address public governor;
+    Pinakion public pinakion;
+    RNG public _RNG;
```

Comment:
Public vars should be underscored. How about `rng`?

Reply:
It's because of the name conflict. The private version would be `__RNG`.

Reply:
It's not a name conflict if its lower case e.g. `Phase public phase;` or `Pinakion public pinakion;`

Reply:
Acronyms are not lower case though.

Reply:
I would prefer to break convention to make an acronym lower case than have a public variable be underscored (I don't think its that unusual). Or would even prefer `randomNumberGenerator`. Double underscoring isn't very distinctive and from the perspective of a caller it is strange to access it with the underscore

Reply:
Changed it to `RNGenerator`. Goes with `RNBlock`.